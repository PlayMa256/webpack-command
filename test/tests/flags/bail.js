const ModuleNotFoundError = require('webpack/lib/ModuleNotFoundError');

const { apply, build, test, validate } = require('../../util');

const fixture = 'bail/bail';
const opts = { fixture };

let config;

test('--bail', module, () => {
  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build, fail, and bail`, () =>
    build(config).catch((error) => {
      expect(error).toEqual(expect.any(ModuleNotFoundError));
    }));
});
