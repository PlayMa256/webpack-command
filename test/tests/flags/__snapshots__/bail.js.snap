// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --bail > should apply #0 1`] = `
Object {
  "bail": true,
  "entry": "<PROJECT_ROOT>/test/fixtures/flags/bail/entry.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
}
`;
